source/app/libpd.c

```diff
@@ -257,7 +269,14 @@ static void parodus_receive()
                         }
                         getCurrentTime(endPtr);
                         WalInfo("Elapsed time : %ld ms\n", timeValDiff(startPtr, endPtr));
-			wrp_free_struct (res_wrp_msg);
+						wrp_free_struct (res_wrp_msg);
+						/* CID-334851 Resource leak fix */
+						if(res_headers != NULL)
+						{
+							WalInfo("Deallocating memory for response headers\n");
+							WAL_FREE(res_headers);
```

Comment:
wrp_free_struct func internally free res headers. So this will cause double free?
https://github.com/xmidt-org/wrp-c/blob/master/src/wrp-c.c#L247